### ISSUE TITLE
Livecharts crash WindowsSdkPackageVersion 10.0.22621.38+

### ISSUE THREAD

**Issue by Bush-cat** [beto-rodriguez/LiveCharts2#1580]
**Describe the bug**
 When loading the Livecharts in XAML in an App with WindowsSdkPackageVersion 10.0.22621.38 or higher it crashes the App and Debugger. WindowsSdkPackageVersion 10.0.22621.34 and lower work as expected

I encountered this when I wanted to update CommunityToolkit.Mvvm from 8.2.2 to 8.3.0, I had to increase WindowsSdkPackageVersion (Microsoft.Windows.CsWinRT 2.1.1 also requires a higher version!)

CommunityToolkit.Mvvm 8.3.0 features many changes like .net8 and other things were made optional.
https://devblogs.microsoft.com/dotnet/announcing-the-dotnet-community-toolkit-830/

**To Reproduce**
1. Set <WindowsSdkPackageVersion>10.0.22621.38</WindowsSdkPackageVersion> in Solution
2. Make a XAML page with a livechart
```xml
<Page
    xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
    xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
    xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
    xmlns:lvc="using:LiveChartsCore.SkiaSharpView.WinUI"
    xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
    mc:Ignorable="d">

    <Grid>
        <lvc:CartesianChart />
    </Grid>
</Page>
```
3. Run the app / open the Page
4. See App and debugger crash, it didn't even save an unhandled exception

**Desktop (please complete the following information):**
 - OS: Windows 11 Build 22631
 - Build 22631

**Additional context**
LiveChartsCore.SkiaSharpView.WinUI 2.0.0-rc2.1
net8.0-windows10.0.22621.0
Microsoft.WindowsAppSDK 1.5.240802000

**Comment by Bush-cat** [comment]
So Windows App SDK 1.6 finally released but it also requires the new SDK

`This version of the Windows App SDK requires Microsoft.Windows.SDK.NET.Ref 10.0.22621.38 or later.`

**Comment by torum** [comment]
yap same here. I got 'System.NotSupportedException' (inside WinRT.Runtime.dll) and 'System.Reflection.TargetInvocationException' (inside System.Private.CoreLib.dll) exceptions after upgrading WinAppSDK to 1.6.

**Comment by beto-rodriguez** [comment]
The problem is that LiveCharts uses the `SKXamlCanvas` from SkiaSharp to render the chart, this control is the one throwing.

Since WinAppSDK 1.6 is relatively new, this is probably something that will be updated soon by someone in the MS team.

Sadly, I am afraid that there not much this library can do here :( we need to wait for MS to update the dependencies of this library.

**Comment by TRadigk** [comment]
Did you address this issue with the devs at WindowsAppSdk at github?

**Comment by beto-rodriguez** [comment]
@TRadigk not sure if the issue is on WindowsAppSdk or SkiaSharp, for now I raised the issue SkiaSharp.

https://github.com/mono/SkiaSharp/issues/3007

**Comment by Bush-cat** [comment]
I guess we need to wait until the Library supports Skiasharp v3 since the new rc version does not work with it yet
`Cannot create instance of type 'LiveChartsCore.SkiaSharpView.WinUI.CartesianChart' `

I’ve been using your Library and appreciate your work. I noticed there was a long gap in updates before the last release candidate and 2.1 and wanted to ask if there could be more frequent updates to the repo or dev branch, at the moment it seems there's only a push when a new release was published.
I'm very interrested in seeing development happen, as this library has a lot of potential but when writing the Issue I got slightly worried the development was stalled.
I don’t mean to pressure you or ask for new builds for every change and I don’t mind force pushes on a dev branch.

**Comment by beto-rodriguez** [comment]
My plan is to maintain the library in the long term.

About frequently updates, normally I do, but as an open source project completely maintained by my resources (mostly time), sometimes things could get slower than usual. The plan is to also offer a paid version that will mitigate this effect.

About SkiaSharp 3, I really hope that this fix also comes to 2.x, because as far as I have read, there will be a lot of breaking changes in the 3.x version, I haven't seen if those changes affect the library, but I am aware that they are working in [making the migration not so painful](https://x.com/SkiaSharp/status/1776308938834677815), the library will be updated to SkiaSharp 3 for sure, when? Well it depends on the effort required to do it.

**Comment by beto-rodriguez** [comment]
> I guess we need to wait until the Library supports Skiasharp v3

Just to be clear, this issue is also present in the latest release of SkiaSharp 3

https://github.com/mono/SkiaSharp/issues/2999

**Comment by Bush-cat** [comment]
I found a reply of the Person who wrote the fix commit about 2.x
> Since that is still preview and SkiaSharp is stable, I cannot really depend on it for SkiaSharp 2.x. However, the 2.x branch does not have the convenient native library that I can use. I technically can add it to 2.x but that is a lot of work, and that old branch is very iffy to build.
https://github.com/microsoft/CsWinRT/issues/1645#issuecomment-2211241512

**Comment by Bush-cat** [comment]
Today I tried building managed Skiasharp 2.x from source for 3hours but it's a big mess, soo many outdated and eol things cause lots of errors, was able to supress and fix some but it's too much, at the moment I am at 237 Errors x_x.

I think I better invest the time in migrating the Library to Skiasharp 3.0 when I get the time, the patched package might not be in nuget yet but you can get it from their azure build server

**Comment by scr47One** [comment]
+1

**Comment by beto-rodriguez** [comment]
This is now working on #1622, (requires the package source `https://aka.ms/skiasharp-eap/index.json`) this will be published when the SkiaSharp fix is also available at NuGet.

**Comment by matthewacme** [comment]
@beto-rodriguez do we need to update the Livecharts at all, or is it just a matter of upgrading the Skiaspharp in the hosting project?

**Comment by matthewacme** [comment]
GREAT NEWS @beto-rodriguez 

I can now confirm, out in the wild, you are absolutely correct #1622 commit fixes this bug and returns your charts to operational status in WinUI running against the 1.6 SDK.

Well done to you and the SkiaSharp team.

Do you happen to have a Patreon or "buy me a coffee" option, I'd like my talk to my Company about giving you some remuneration for your hard work.

**Comment by beto-rodriguez** [comment]
@matthewacme glad to know it is working for you also!

> Do you happen to have a Patreon or "buy me a coffee" option, I'd like my talk to my Company about giving you some remuneration for your hard work.

Not for now, but the library will offer a paid version in a future version, thanks for interest in supporting my work!

**Comment by matthewacme** [comment]
You just saved my company 1000s of dollars in purchasing licenses for a commercial charting product (we were looking at Telerik and Syncfusion).

I'd like to suggest it might be worth your while to set up a Patreon account. I think I can make a good case that we should donate 10% of our savings.

Keep up the good work!


[cid:230eee75-33b0-495e-9e07-46031e4c0f76]
Matthew C. Hintzen [he-him-his] | Windows Washer
•
w – ***@***.******@***.***>
p – 321.234.8839
•
________________________________

**Comment by ArpitViradiyaCrowncon** [comment]
Can anyone help me for how to set/configure package source "https://aka.ms/skiasharp-eap/index.json". I'm using in .Net maui.

**Comment by jpageacsys** [comment]
> Can anyone help me for how to set/configure package source "https://aka.ms/skiasharp-eap/index.json". I'm using in .Net maui.

<img width="551" alt="image" src="https://github.com/user-attachments/assets/7d34b1ee-3fc1-4574-9776-72219fa27e8d">

I believe you can add it like this in NuGet settings in Visual Studio

**Comment by SiL-RHU** [comment]
I there, 

Face the same issue when updating the CommunityToolkit.Mvvm to version 8.3.2 and try to generate the packages from the skiasharp-3 branch but unfortunately without any success.

After cloning the repo locally, switch to the skiasharp-3 and adapted the projects configurations to only build the WinUi version of the package. 

When using the provided scripts, I'm able to generate the packages but when I use them in our application, I received the following exception:
> Exception thrown: 'Microsoft.UI.Xaml.Markup.XamlParseException' in WinRT.Runtime.dll
WinRT information: Cannot locate resource from 'ms-appx:///LiveChartsCore.SkiaSharpView.WinUI/CartesianChart.xaml'.
XAML parsing failed.

I then tried to generate the package from Visual Studio 17.11.5 but get the following build error in Release:
> src\skiasharp\LiveChartsCore.SkiaSharp\VisualElements\SVGVisual.cs(45,38,45,51): error CS0507: 'SVGVisual.OnInvalidated(Chart<SkiaSharpDrawingContext>)': cannot change access modifiers when overriding 'protected internal' inherited member 'GeometryVisual<SVGPathGeometry, LabelGeometry, SkiaSharpDrawingContext>.OnInvalidated(Chart<SkiaSharpDrawingContext>)'

Switching back the configuration to Debug solved this issue but lead to the first exception when generating the packages.

Do you have any advice on how to generate a package that contains the fix?

**Comment by core2lord** [comment]
> @SiL-RHU to make it simpler and avoid building your own NuGet packages, I recommend you remove the reference to LiveCharts packages from NuGet then directly add the reference to LiveChartsCore.SkiaSharpView.Maui it is under the `src/skiasharp` folder.

Would this work for 'LiveChartsCore.SkiaSharpView.WinUI' as well? Unfortunately, I wasn't able to figure out getting #1622 to work and I guess I broke VS2022 enough that even reverting back to 17.11.4 crashes identically, probably now requiring reinstall, but was curious if this is something I could try afterwords.

**Comment by beto-rodriguez** [comment]
> Would this work for 'LiveChartsCore.SkiaSharpView.WinUI' as well? Unfortunately, I wasn't able to figure out getting #1622 to work and I guess I broke VS2022 enough that even reverting back to 17.11.4 crashes identically, probably now requiring reinstall, but was curious if this is something I could try afterwords.

It should work also, you can try before reinstalling vs .

Also if you re install the previous visual  studio version you will not face this issues on Maui.

**Comment by frostshoxx** [comment]
> > Can anyone help me for how to set/configure package source "https://aka.ms/skiasharp-eap/index.json". I'm using in .Net maui.
> 
> <img alt="image" width="551" src="https://private-user-images.githubusercontent.com/69006582/375847815-7d34b1ee-3fc1-4574-9776-72219fa27e8d.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjk2MTgxOTIsIm5iZiI6MTcyOTYxNzg5MiwicGF0aCI6Ii82OTAwNjU4Mi8zNzU4NDc4MTUtN2QzNGIxZWUtM2ZjMS00NTc0LTk3NzYtNzIyMTlmYTI3ZThkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMjIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDIyVDE3MjQ1MlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTc4MTQ1MDE0YTVhZWQyZmQ0ZWJiYWQwZGZjMTExZGFiOTM3ODNhNzU3MzMzOTgxYmU2YTgzZTY3ZmFjMjk5NTgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.EG-eCnXrdut0t5CWdHJNzDFbFaJAO93Zo9900_7YhHo">
> I believe you can add it like this in NuGet settings in Visual Studio

On the manage package for solution page, don't forget to select the source to skia sharp (however you name it in the settings) and select "Include Prerelease"
![image](https://github.com/user-attachments/assets/3c839a1b-496d-45f9-8da5-ed85fa1222a0)

**Comment by beto-rodriguez** [comment]
This is now working using the latest version 2.0.0-rc4, I will close this for now, feel free to reply if something is not working.

**Comment by JuandreG** [comment]
I added LiveChartsCore.SkiaSharpView.WinUI 2.0.0-rc5 to my project and when I try to build a Gauge nothing shows up on the page. I didn't see any exceptions and the application does not break, there is just nothing showing. When I update to 2.0.0-rc5.1 I get the exception that is in the title of this issue.

I am using Microsoft.WindowsAppSDK 1.6.250108002 and Microsoft.Windows.SDK.BuildTools 10.0.26100.1742.

I also tried LiveChartsCore.SkiaSharpView.WinUI 2.0.0-rc4.5 and same issue, nothing shows up.

**Comment by JuandreG** [comment]
I am trying to use this sample: 

https://livecharts.dev/docs/WinUi/2.0.0-rc5/samples.pies.angularGauge

I was mistaken, on LiveChartsCore.SkiaSharpView.WinUI 2.0.0-rc5 I see this exception when the page calls InitializeComponent:

![Image](https://github.com/user-attachments/assets/8bbe10e7-a16f-45d5-8d78-a35b13beed8e)

**Comment by Strypper** [comment]
Hey guys what's the update on this? I'm using WinUI3 1.7 and it crash in Release mode

**Comment by Bush-cat** [comment]
> Hey guys what's the update on this? I'm using WinUI3 1.7 and it crash in Release mode

This issue has been solved already, my application running 1.7 is working.

**Comment by frostshoxx** [comment]
> Hey guys what's the update on this? I'm using WinUI3 1.7 and it crash in Release mode

Try 2.0.0-rc4.5 or newer. It's been working for me as well.
